lib/chef/provider/windows_task.rb

```
@@ -52,40 +52,58 @@ def set_current_resource(task_hash)
-          options["D"] = @new_resource.day if @new_resource.day
-          options["M"] = @new_resource.months unless @new_resource.months.nil?
-
-          run_schtasks "CREATE", options
-          xml_options = []
-          xml_options << "cwd" if new_resource.cwd
-          xml_options << "random_delay" if new_resource.random_delay
-          xml_options << "execution_time_limit" if new_resource.execution_time_limit
-          update_task_xml(xml_options) unless xml_options.empty?
+      # This method checks if task and command attributes exist since those two are mandatory attributes to create a schedules task.
+      def basic_validation
+        validate = []
+        validate << "Command" if @new_resource.command.nil? || @new_resource.command.empty?
+        validate << "Task Name" if @new_resource.task_name.nil? || @new_resource.task_name.empty?
+        return true if validate.empty?
+        Chef::Log.error "Value for '#{validate.join(', ')}' option cannot be empty"
+        false
+      end
 
-          new_resource.updated_by_last_action true
-          Chef::Log.info "#{@new_resource} task created"
+      def action_create
+        if @current_resource.exists
+          if !(task_need_update? || @new_resource.force)
+            Chef::Log.info "#{@new_resource} task already exists - nothing to do"
+            return
+          end
+          # To merge current resource and new resource attributes
+          temp_arry = %w{ command user run_level cwd frequency_modifier frequency idle_time random_delay execution_time_limit start_day start_time }
+          temp_arry.each do |attribute|
+            new_resource_attribute = @new_resource.send(attribute)
+            current_resource_attribute = @current_resource.send(attribute)
+            @new_resource.send("#{attribute}=", current_resource_attribute ) if current_resource_attribute && new_resource_attribute.nil?
+          end
         end
+        return unless basic_validation
```

Do we think this should raise an error since the user has asked us to create a resource but we can't?